samples/customABSample/src/main/java/com/inmobi/customabsample/BannerCustomABActivity.java

```diff
@@ -0,0 +1,138 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_banner_custom_a_b);
+        preload = findViewById(R.id.preload);
+        preload.setOnClickListener(this);
+        show = findViewById(R.id.show);
+        show.setOnClickListener(this);
+
+        setupBannerAd();
+    }
+
+    private void setupBannerAd() {
+        mInMobiBanner = new InMobiBanner(BannerCustomABActivity.this, PlacementId.YOUR_BANNER_PLACEMENT_ID);
+        RelativeLayout adContainer = (RelativeLayout) findViewById(R.id.ad_container);
+        mInMobiBanner.setAnimationType(InMobiBanner.AnimationType.ROTATE_HORIZONTAL_AXIS);
+        mInMobiBanner.setEnableAutoRefresh(false);
+        mInMobiBanner.setListener(new BannerAdEventListener() {
+            @Override
+            public void onAdLoadSucceeded(@NonNull InMobiBanner inMobiBanner,
+                                          @NonNull AdMetaInfo adMetaInfo) {
+                super.onAdLoadSucceeded(inMobiBanner, adMetaInfo);
+                show.setVisibility(View.INVISIBLE);
+                Log.d(TAG, "onAdLoadSucceeded");
+            }
+
+            @Override
+            public void onAdLoadFailed(InMobiBanner inMobiBanner, InMobiAdRequestStatus inMobiAdRequestStatus) {
+                super.onAdLoadFailed(inMobiBanner, inMobiAdRequestStatus);
+                show.setVisibility(View.INVISIBLE);
+                Log.d(TAG, "Banner ad failed to load with error: " +
+                        inMobiAdRequestStatus.getMessage());
+            }
+
+            @Override
+            public void onAdClicked(InMobiBanner inMobiBanner, Map<Object, Object> map) {
+                super.onAdClicked(inMobiBanner, map);
+                Log.d(TAG, "onAdClicked");
+            }
+
+            @Override
+            public void onAdDisplayed(InMobiBanner inMobiBanner) {
+                super.onAdDisplayed(inMobiBanner);
+                Log.d(TAG, "onAdDisplayed");
+            }
+
+            @Override
+            public void onAdDismissed(InMobiBanner inMobiBanner) {
+                super.onAdDismissed(inMobiBanner);
+                Log.d(TAG, "onAdDismissed");
+            }
+
+            @Override
+            public void onUserLeftApplication(InMobiBanner inMobiBanner) {
+                super.onUserLeftApplication(inMobiBanner);
+                Log.d(TAG, "onUserLeftApplication");
+            }
+
+            @Override
+            public void onRewardsUnlocked(InMobiBanner inMobiBanner, Map<Object, Object> map) {
+                super.onRewardsUnlocked(inMobiBanner, map);
+                Log.d(TAG, "onRewardsUnlocked");
+            }
+
+            @Override
+            public void onAdFetchSuccessful(@NonNull InMobiBanner inMobiBanner, @NonNull AdMetaInfo adMetaInfo) {
+                super.onAdFetchSuccessful(inMobiBanner, adMetaInfo);
+                Log.d(TAG, "onAdFetchSuccessful");
```

Comment:
Purpose of using custom AB is to get a Bid at right point. please show case how to get the adMetaInfo and bid